### DISCUSSION TITLE
which project should we set as startup project while running the application and also when can we expect the proper documentation of the this repo

### DISCUSSION THREAD

**Discussion by devdgupta** [fullstackhero/dotnet-microservices-boilerplate#68] [Q&A] [answered]

**Comment by devdgupta** [comment]
I am simple using tye run --dashboard to run the application. Is this correct way of running the microservices application. from where I need to generate the token so that i can use while calling the api

**Comment by iammukeshm** [comment, marked as answer]
If you check the current Readme of the repo, you can find the instructions,

- Run `make tye` inside the ./fluentpos folder, to run all the services of fluentpos.
- Identity Service also starts up, which can generate tokens for you, the sample API calls are already included in the thunder-tests folder. You can install the Thunder client extension on your Visual Code, and browse it from the solution's root. 

However, here is the CURL command to generate tokens, in case

```
curl -X POST \
  'https://localhost:7001/connect/token' \
  --header 'Accept: */*' \
  --header 'User-Agent: Thunder Client (https://www.thunderclient.com)' \
  --header 'Content-Type: application/x-www-form-urlencoded' \
  --data-urlencode 'grant_type=client_credentials' \
  --data-urlencode 'client_id=console' \
  --data-urlencode 'client_secret=858b39fd-3908-45cb-ab14-aa58220f6d69' \
  --data-urlencode 'scope=catalog:read cart:read'
```

Another approach by using the docker-compose is also mentioned in the Readme.md file.

The project is currently at its early stage, where I am trying to build a framework and use it in a real-world application (fluentpos). However, the project is taking good shape now. I still have to cover a couple of topics/concepts like
- service invocations
- cloud deployment with terraform
- event sourcing
- more structural cleanup
- authentication models
- maybe a blazor / nextjs web app (at the end for fluentpos)

Once these are done, documentations will be started along with a walkthrough video.

**Reply by Asimkhan-developer** [reply]
Is the work still in progress for this boilerplate? 
Can we use it in our project to startup ?
